### PR TITLE
Proposed resolution for the L1CSM YANG model review comments

### PR DESCRIPTION
**Pull request by italobusi**
Proposed resolution for the L1CSM YANG model review comments (#64)

### CODE REVIEW

**Comment by italobusi** [comment]
Positive feedbacks received from Netmod WG discussion about the possibility to define identities with multiple bases (e.g., STM-1 as being based on both coding-function and client-signal):

https://mailarchive.ietf.org/arch/msg/netmod/LpiyIo9tmfX6G8De7UENFWWJDbw/

**Comment by italobusi** [comment]
Triggered discussion on-going in Netmod WG about the possibility to use identities with multiple bases to check consistency between protocol, coding-function and optical-interface-functions:

https://mailarchive.ietf.org/arch/msg/netmod/BPl5GQpMPNFeAeDw2ueVqxZCPO0/

Discussion still on-going

**Comment by sergiobelotti** [comment]
> Positive feedbacks received from Netmod WG discussion about the possibility to define identities with multiple bases (e.g., STM-1 as being based on both coding-function and client-signal):
> 
> https://mailarchive.ietf.org/arch/msg/netmod/LpiyIo9tmfX6G8De7UENFWWJDbw/

@italobusi : so, taking your example, it means Option #1 ?
> >> > The value space is any identity which is derived from both base identities.
> >> This means the only valid value in the example would be 'b'.

**Comment by sergiobelotti** [comment]
> Triggered discussion on-going in Netmod WG about the possibility to use identities with multiple bases to check consistency between protocol, coding-function and optical-interface-functions:
> 
> https://mailarchive.ietf.org/arch/msg/netmod/BPl5GQpMPNFeAeDw2ueVqxZCPO0/
> 
> Discussion still on-going

@italobusi : I'm confused, is it not the "positive answer" on the previous comment?

**Comment by agva123** [comment]
Appended one more change to fix the relative path in when statement which caused compilation warnings. Please review

**Comment by agva123** [comment]
I added an "otn" container in the otn-path-bandwidth grouping and changed to use derived-from-or-self() to remove yanglint warnings.

**Comment by italobusi** [comment]
> > Triggered discussion on-going in Netmod WG about the possibility to use identities with multiple bases to check consistency between protocol, coding-function and optical-interface-functions:
> > https://mailarchive.ietf.org/arch/msg/netmod/BPl5GQpMPNFeAeDw2ueVqxZCPO0/
> > Discussion still on-going
> 
> @italobusi : I'm confused, is it not the "positive answer" on the previous comment?

From the answer it is still not clear how to write the must statement to check the consistency.

In order to move forward, I have added the consistency requirement in the description and created open issue #72 to further investigate this issue (mainly in the scope of L1CSM draft than of layer1-types draft).

File: YANG/ccamp/l1csm/ietf-l1csm.tree [hunk 1]
```diff
@@ -3,10 +3,14 @@ module: ietf-l1csm
...
-     |        +--rw protocol?            identityref
-     |        +--rw coding?              identityref
-     |        +--rw optical-interface?   identityref
+     |        +--rw id                         string
+     |        +--rw (uni-access-type)?
+     |           +--:(mef)
+     |           |  +--rw protocol             identityref
+     |           |  +--rw coding               identityref
+     |           |  +--rw optical-interface    identityref
+     |           +--:(itu)
```
Comment (haomianzheng) [inline YANG/ccamp/l1csm/ietf-l1csm.tree:12]: Yes ITU is different with MEF, but has the same as IETF UNI. Shall we put the branch name as 'itu or ietf'? or something else to cover broader scope?

Comment (haomianzheng) [inline YANG/ccamp/l1csm/ietf-l1csm.tree:12]: Yes ITU is different with MEF, but has the same as IETF UNI. Shall we put the branch name as 'itu or ietf'? or something else to cover broader scope?

File: YANG/ccamp/l1csm/ietf-l1csm.yang [hunk 2]
```diff
@@ -162,7 +216,26 @@ module ietf-l1csm {
...
+          choice uni-access-type {
+            description
+              "The UNI access type can be specified either by the
+               protocol, coding function and optical interface
+               function, defined in MEF, or by the client-signal,
+               defined in ITU-T.";
+            case mef {
+              uses protocol-coding-optical-interface;
+            }
+            case itu {
```
Comment (haomianzheng) [inline YANG/ccamp/l1csm/ietf-l1csm.yang:228]: Same Above.

**Review by haomianzheng — APPROVED** [review]
The work is ready to go (with minor changes as proposed in the review).

File: YANG/ccamp/l1csm/ietf-l1csm.tree (hunk 1 above)
Comment (haomianzheng) [reply YANG/ccamp/l1csm/ietf-l1csm.tree:12]: It may introduce other SDO to make the branch complicated. Let's keep as is.

**Pull request merged**